src/rgw/rgw_lua_background.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include "common/dout.h"
+#include "rgw_common.h"
+#include <string>
+#include "rgw_lua_utils.h"
+
+namespace rgw::lua {
+
+//Interval between each execution of the script is set to 5 seconds
+constexpr const int INIT_EXECUTE_INTERVAL = 5;
+
+//Writeable meta table named RGW with mutex protection
+using BackgroundMap = std::unordered_map<std::string, std::string>;
+struct RGWTable : StringMapMetaTable<BackgroundMap,
+  StringMapWriteableNewIndex<BackgroundMap>> {
+    static std::string TableName() {return "RGW";}
+    static std::string Name() {return TableName() + "Meta";}
+    static int IndexClosure(lua_State* L) {
+      const auto mtx = reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
```

Comment:
```suggestion
      auto& mtx = *reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
      std::lock_guard l(mtx);
```
you want a reference to the same mutex, not a new one
also, the mutex is not const